src/imap/select_folder.rs

```diff
@@ -19,35 +19,31 @@ pub enum Error {
     #[error("Got a NO response when trying to select {0}, usually this means that it doesn't exist: {1}")]
     NoFolder(String, String),
 
-    #[error("IMAP close/expunge failed")]
-    CloseExpungeFailed(#[from] async_imap::error::Error),
-
     #[error("IMAP other error: {0}")]
     Other(String),
 }
 
+impl From<anyhow::Error> for Error {
+    fn from(err: anyhow::Error) -> Error {
+        Error::Other(err.to_string())
```

Comment:
`to_string()` probably formats with `{}`, so it'd be better to use `Error::Other(format!("{:#}", err))`